### ISSUE TITLE
The rule "Present" not working as intended

### ISSUE THREAD

**Issue by Squad-G** [spatie/laravel-data#488]
**✏️ Describe the bug**
Laravel's documentation state the following: 
[present](https://laravel.com/docs/10.x/validation#rule-present)
The field under validation must exist in the input data.

Laravel validator is throwing an error: "The field contact.notes is required."

**↪️ To Reproduce**
**JSON Input:**

```
{
    "localeId":"fr_CA",
    "email":"ffdftddfdssgdsd@example.com",
    "type":"employee",
    "password":"12345678abcd",
    "passwordConfirmation":"12345678abcd",
    "contact": {
        "prefix": "",
        "firstName": "fd",
        "lastName": "fds",
        "suffix": "sr",
        "notes": ""
    }
}
```

**PHP Classes**
```
final class UserCreateRequestDto extends Data
{
	/**
	 * Create a new data instance from the given arguments.
	 * 
	 * @param string $user_email The user email
	 * @param LocaleEnum $locale_id The locale id
	 * @param UserTypeEnum $user_type The user type
	 * @param string $user_password The user password
	 * @param string $user_password_confirmation The user password confirmation
	 * @param UserCreateContactRequestDto $contact The contact
	 *
	 * @return void Returns nothing
	 */
	public function __construct(
		#[MapInputName('email'), Required, StringType, Rule(new EmailValidationRule(120, true)), Unique(User::class, 'user_email')]
		public string $user_email,

		#[MapInputName('localeId'), Required, StringType, Enum(LocaleEnum::class)]
		public LocaleEnum $locale_id,
		
		#[MapInputName('type'), Required, StringType, Enum(UserTypeEnum::class)]
		public UserTypeEnum $user_type,

		#[MapInputName('password'), Required, StringType, Password(default: true)]
		public string $user_password,

		#[MapInputName('passwordConfirmation'), Required, StringType, Same('password')]
		public string $user_password_confirmation,

		#[MapInputName('contact'), Present, Nullable]
		public null|UserCreateContactRequestDto $contact
	) { }
}

final class UserCreateContactRequestDto extends Data
{
	/**
	 * Create a new data instance from the given arguments.
	 * 
	 * @param null|ContactPrefixEnum $ontact_prefix The contact prefix
	 * @param string $contact_first_name The contact first name
	 * @param string $contact_last_name The contact last name
	 * @param null|ContactSuffixEnum $contact_suffix The contact suffix
	 * @param string $contact_notes The contact notes
	 *
	 * @return void Returns nothing
	 */
	public function __construct(
		#[MapInputName('prefix'), Present, Nullable, StringType, Enum(ContactPrefixEnum::class)]
		public null|ContactPrefixEnum $contact_prefix,

		#[MapInputName('firstName'), Required, StringType, Max(250)]
		public string $contact_first_name,
		
		#[MapInputName('lastName'), Required, StringType, Max(250)]
		public string $contact_last_name,

		#[MapInputName('suffix'), Present, Nullable, StringType, Enum(ContactSuffixEnum::class)]
		public null|ContactSuffixEnum $contact_suffix,

		#[MapInputName('notes'), Present, StringType, Max(65535)]
		public string $contact_notes
	) { }
}
```

**✅ Expected behavior**
Present should not throw an error here because the field is indeed in the request.

**🖥️ Versions**

Laravel: 10.4.1
Laravel Data:
PHP: 8.2.5

**Comment by rubenvanassche** [comment]
I think this will be fixed with data 3.7.0 which we'll release later today, could you check again with that version?

**Comment by Squad-G** [comment]
I have upgraded the package to 3.7.0 and it appears to have been fixed indeed.
